File: src/main/java/ru/timakden/chesspuzzle/ChessPieceFactory.java
package ru.timakden.chesspuzzle;

/**
 * Фабрика для создания шахматных фигур.
 */
class ChessPieceFactory {
    ChessPiece getChessPiece(String chessPieceType, int row, int column) {
        switch (chessPieceType) {
            case "K":
                return new King(row, column);
            case "R":
                return new Rook(row, column);
            case "N":
                return new Knight(row, column);
            case "Q":
                return new Queen(row, column);
            case "B":
                return new Bishop(row, column);
            default:
                return null;
        }
    }
}

File: src/main/kotlin/ru/timakden/chesspuzzle/ChessPieceFactory.kt
package ru.timakden.chesspuzzle

object ChessPieceFactory {
    fun getPiece(pieceType: String, row: Int, column: Int): ChessPiece {
        return when (pieceType) {
            KING -> King(row, column)
            ROOK -> Rook(row, column)
            KNIGHT -> Knight(row, column)
            QUEEN -> Queen(row, column)
            BISHOP -> Bishop(row, column)
            else -> throw IllegalArgumentException("Unknown chess piece type: $pieceType")
        }
    }
}
